x-pack/platform/plugins/shared/rule_registry/server/routes/get_alert_fields_by_rule_type_ids.ts

```diff
@@ -0,0 +1,154 @@
+        );
+
+        const internalUserEsClient = (await context.core).elasticsearch.client.asInternalUser;
+        const currentUserEsClient = (await context.core).elasticsearch.client.asCurrentUser;
+
+        const registeredRuleTypes = racContext.alerting.listTypes();
+
+        // fetch all rule types if no specific rule type Ids are provided
+        const allRuleTypesIds = ruleTypeIds.length
+          ? Array.isArray(ruleTypeIds)
+            ? ruleTypeIds
+            : [ruleTypeIds]
+          : Array.from(registeredRuleTypes.keys());
+
+        const authorizedRuleTypes = await authorization.getAllAuthorizedRuleTypesFindOperation({
+          authorizationEntity: AlertingAuthorizationEntity.Alert,
+          ruleTypeIds: allRuleTypesIds,
+        });
+
+        const authorizedRuleTypesIds = Array.from(authorizedRuleTypes.keys());
+
+        const siemRuleTypeIds = authorizedRuleTypesIds.filter((ruleTypeId) =>
+          isSiemRuleType(ruleTypeId)
+        );
+        const siemIndices =
+          (siemRuleTypeIds ? await alertsClient.getAuthorizedAlertsIndices(siemRuleTypeIds) : []) ??
+          [];
+
+        const otherRuleTypeIds = authorizedRuleTypesIds.filter(
+          (ruleTypeId) => !isSiemRuleType(ruleTypeId)
+        );
+        const otherIndices =
+          (otherRuleTypeIds
+            ? await alertsClient.getAuthorizedAlertsIndices(otherRuleTypeIds)
+            : []) ?? [];
+
+        const indexFilter = {
+          range: {
+            '@timestamp': {
+              gte: 'now-90d',
+            },
+          },
+        };
+
+        const dataViewsService = await racContext.dataViews.dataViewsServiceFactory(
+          racContext.savedObjectsClient,
+          currentUserEsClient
+        );
+
+        // Fetch fields for SIEM indices using current user client
+        const siemFields = siemIndices.length
+          ? await dataViewsService.getFieldsForWildcard({
+              pattern: siemIndices.join(','),
+              allowNoIndex: true,
+              includeEmptyFields: false,
+              indexFilter,
+            })
+          : [];
+
+        const indexPatternsFetcherAsInternalUser = new IndexPatternsFetcher(internalUserEsClient);
+
+        // Fetch fields for other indices using internal user client
+        const { fields: otherFields } = otherIndices.length
+          ? await indexPatternsFetcherAsInternalUser.getFieldsForWildcard({
+              pattern: otherIndices,
+              metaFields: ['_id', '_index'],
+              fieldCapsOptions: { allow_no_indices: true },
+              includeEmptyFields: false,
+              indexFilter,
+            })
+          : { fields: [] };
+
+        const uniqueFields = mergeUniqueFieldsByName(otherFields, siemFields);
+
+        const mappedFields = {
+          alertFields: fieldDescriptorToBrowserFieldMapper(uniqueFields),
+          fields: uniqueFields,
+        };
+
+        return response.ok({
+          body: mappedFields,
```

Comment:
the response should also be versioned

Reply:
[Versioned](https://github.com/elastic/kibana/pull/222032/commits/3f0456483d18124249ed3b43a7df89bd015e75e4#diff-d1703ab9db0fa6164443d46f120683d9b12ff1f7cbdb42ce90ccfe22f2721699)